apollo-converters/pojo/src/test/java/com/apollographql/android/converter/pojo/IntegrationTest.java

```diff
@@ -35,14 +39,35 @@
   interface Service {
     @POST("graphql")
     Call<Response<AllPlanets.Data>> heroDetails(@Body GraphQlOperationRequest<AllPlanets.Variables> query);
+
+    @POST("graphql")
+    Call<Response<ProductsWithDate.Data>> productsWithDate(@Body GraphQlOperationRequest<ProductsWithDate.Variables>
+        query);
   }
 
   @Rule public final MockWebServer server = new MockWebServer();
 
   @Before public void setUp() {
+    CustomTypeAdapter<Date> dateCustomTypeAdapter = new CustomTypeAdapter<Date>() {
+      final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
+
+      @Override public Date decode(String value) {
+        try {
+          return dateFormat.parse(value);
+        } catch (ParseException e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      @Override public String encode(Date value) {
+        return null;
+      }
+    };
+
     Retrofit retrofit = new Retrofit.Builder()
         .baseUrl(server.url("/"))
-        .addConverterFactory(new ApolloConverterFactory())
+        .addConverterFactory(new ApolloConverterFactory()
```

Comment:
I feel like the api would be nicer if I can do `.addConverterFactory(ApolloConverter.builder().withCustomTypeAdapter().build())`